gcc/rust/typecheck/rust-hir-type-check-pattern.cc

```diff
@@ -466,8 +466,15 @@ TypeCheckPattern::typecheck_range_pattern_bound (
 void
 TypeCheckPattern::visit (HIR::AltPattern &pattern)
 {
-  rust_sorry_at (pattern.get_locus (),
-		 "type checking alternate patterns not supported");
+  const auto &alts = pattern.get_alts ();
+
+  for (auto &alt_pattern : alts)
+    {
+      // do something with the result?
+      auto result = TypeCheckPattern::Resolve (alt_pattern.get (), parent);
```

Comment:
What are the rules with alt patterns are the types all meant to match? We do need a least upper bound coercion helper which is missing at the moment but you could copy what we do for array expressions at least for now and raise an issue and a TODO comment.

Ideally it should look like this:
```
std::vector<TyTy::BaseType*> alts;

for each pattern {
auto result = TypeCheckPattern::Resolve (alt_pattern.get (), parent);
  alts.push(result);

}

lub_coercion(alts, pattern.get_locus (), pattern.get_mappings().get_hirid())
```

But yeah that lub_coercion helper doesnt exist right now. See:  https://github.com/Rust-GCC/gccrs/blob/8e2e7fa0211f49fe8f4c8b156468320ccfa711a0/gcc/rust/typecheck/rust-hir-type-check-expr.cc#L933C9-L943

its not a perfect lub, it should really be coercion site but its ok for now.